### PR TITLE
Modify ReferenceExpression to hold topLevelParent & Netsuite support references

### PR DESCRIPTION
**Pull request by omrilit**

### CODE REVIEW

**Comment by omrilit** [comment]
Please review this PR commit by commit as I intend to push them separately (no squash)

File: packages/netsuite-adapter/src/filters/instance_references.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+const replaceReferenceValues = (
+  values: Values,
+  refElement: ObjectType,
+  serviceIdToElemID: Record<string, ElemID>
+): Values => {
+  const replacePrimitive: TransformFunc = ({ value }) => {
+    if (!_.isString(value)) {
+      return value
+    }
+    const capturedServiceId = value.match(pathReferenceRegex)?.groups?.[CAPTURE]
```
Comment (roironn) [inline packages/netsuite-adapter/src/filters/instance_references.ts:86]: I would probably move this to. a separate function or add doc just to make it clear what exactly we are catching here.

File: packages/netsuite-adapter/src/filters/instance_references.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import {
+  SUITE_SCRIPTS_FOLDER_NAME, TEMPLATES_FOLDER_NAME, WEB_SITE_HOSTING_FILES_FOLDER_NAME, SCRIPT_ID,
+  PATH,
+} from '../constants'
+import { serviceId } from '../transformer'
+import { FilterCreator } from '../filter'
+import { isCustomType } from '../types'
+
+const CAPTURE = 'capture'
+const pathReferenceRegex = new RegExp(`^\\[(?<${CAPTURE}>\\/(${TEMPLATES_FOLDER_NAME}|${SUITE_SCRIPTS_FOLDER_NAME}|${WEB_SITE_HOSTING_FILES_FOLDER_NAME})\\/.+)]$`)
```
Comment (roironn) [inline packages/netsuite-adapter/src/filters/instance_references.ts:33]: Maybe add docs with example for the values the regexes suppose to catch?

File: packages/netsuite-adapter/src/transformer.ts [hunk 3]
```diff
@@ -250,5 +251,31 @@ export const toCustomizationInfo = (instance: InstanceElement): CustomizationInf
...
+const getScriptIdParts = (topLevelParent: InstanceElement, elemId: ElemID): string[] => {
+  if (elemId.isTopLevel()) {
+    return [topLevelParent.value[SCRIPT_ID]]
+  }
+  const relativePath = elemId.createTopLevelParentID().path
+  const value = _.get(topLevelParent.value, relativePath)
+  if (_.has(value, SCRIPT_ID)) {
+    return [...getScriptIdParts(topLevelParent, elemId.createParentID()), value[SCRIPT_ID]]
+  }
+  return getScriptIdParts(topLevelParent, elemId.createParentID())
```
Comment (roironn) [inline packages/netsuite-adapter/src/transformer.ts:266]: Why do we want the last return value? Seems like this should never be called right?

Comment (omrilit) [reply packages/netsuite-adapter/src/transformer.ts:266]: This method is recursive and we start by calling to the deepest field and then climb up until we reach the top level instance's SCRIPT_ID.
On the way up, not all fields have the SCRIPT_ID field directly, and thus we should still climb up until we'll find such. Please LMK if it's still unclear

**Pull request merged**